enterprise/views.py

```diff
@@ -472,6 +475,81 @@ def post(self, request):
         return self.post_account_consent(request, consent_provided)
 
 
+class HandleConsentEnrollment(View):
+    """
+    Handle enterprise course enrollment at providing data sharing consent.
+
+    View handles the case for enterpries course enrollment after successful
+    consent.
+    """
+
+    @method_decorator(enterprise_login_required)
+    def get(self, request, enterprise_uuid, course_id):
+        """
+        Handle the enrollment of enterprise learner in the provided course.
+
+        Based on `enterprise_uuid` in URL, the view will decide which
+        enterprise customer's course enrollment record should be created.
+
+        Depending on the value of query parameter `course_mode` then learner
+        will be either redirected to LMS dashboard for audit modes or
+        redirected to ecommerce basket flow for payment of premium modes.
+        """
+        # Verify that all necessary resources are present
+        verify_edx_resources()
+        enrollment_course_mode = request.GET.get('course_mode')
+
+        # Redirect the learner to LMS dashboard in case no course mode is
+        # provided as query parameter `course_mode`
+        if not enrollment_course_mode:
+            return redirect(LMS_DASHBOARD_URL)
+
+        try:
+            enrollment_client = EnrollmentApiClient()
+            course_modes = enrollment_client.get_course_modes(course_id)
+        except HttpClientError:
+            logger.error('Failed to determine available course modes for course ID: %s', course_id)
+            raise Http404
+
+        # Verify that the request user belongs to the enterprise against the
+        # provided `enterprise_uuid`.
+        enterprise_customer = get_enterprise_customer_or_404(enterprise_uuid)
+        enterprise_customer_user = get_enterprise_customer_user(request.user.id, enterprise_customer.uuid)
+        if not enterprise_customer_user:
+            raise Http404
+
+        selected_course_mode = None
+        for course_mode in course_modes:
+            if course_mode['slug'] == enrollment_course_mode:
+                selected_course_mode = course_mode
+                break
+
+        if not selected_course_mode:
+            return redirect(LMS_DASHBOARD_URL)
+
+        # Create the Enterprise backend database records for this course
+        # enrollment
+        EnterpriseCourseEnrollment.objects.get_or_create(
+            enterprise_customer_user=enterprise_customer_user,
+            course_id=course_id,
```

Comment:
We don't seem to be setting the consent flag on the record; I believe that'd be necessary here.